okhttp/src/main/java/io/grpc/okhttp/AsyncSink.java

```diff
@@ -157,15 +159,26 @@ public void run() {
   }
 
   private abstract class WriteRunnable implements Runnable {
+
+    private final Link link;
+
+    WriteRunnable() {
+      this.link = PerfMark.linkOut();
```

Comment:
The link should be created within a started task. Also, I do not think this is the best place to create links.

Reply:
Done.